### PR TITLE
Merkle proofs and Updates batches

### PR DESCRIPTION
**Pull request by mateuszaaa**
## New storage entries
Rolldown:
- L2RequestsBatch - stores every created batch, grouped by chain
- L2RequestsBatchLast - stores last created batch per chain

## New rpc
```
	#[method(name = "rolldown_get_merkle_root")]
	fn get_merkle_root(
		&self,
		chain: Chain,
		range: (u128, u128),
		at: Option<BlockHash>,
	) -> RpcResult<H256>;

	#[method(name = "rolldown_get_merkle_proof")]
	fn get_merkle_proof(
		&self,
		chain: Chain,
		range: (u128, u128),
		tx_id: u128,
		at: Option<BlockHash>,
	) -> RpcResult<Vec<H256>>;

	#[method(name = "rolldown_verify_merkle_proof")]
	fn verify_merkle_proof(
		&self,
		chain: Chain,
		range: (u128, u128),
		tx_id: u128,
		root: H256,
		proof: Vec<H256>,
		at: Option<BlockHash>,
	) -> RpcResult<bool>;

```

## New Behaviour
- automatic batch creation after configurable amount of l2 request
- automatic batch creation after configurable amount of time(blocks) since last update

### CODE REVIEW

File: pallets/rolldown/rpc/src/lib.rs [hunk 1]
```diff
@@ -46,6 +47,34 @@ pub trait RolldownApi<BlockHash, L1Update, Chain> {
...
+	#[method(name = "rolldown_verify_merkle_proof")]
+	fn verify_merkle_proof(
+		&self,
+		chain: Chain,
+		range: (u128, u128),
+		tx_id: u128,
+		root: H256,
+		proof: Vec<H256>,
+		at: Option<BlockHash>,
+	) -> RpcResult<bool>;
```
Comment (mateuszaaa) [inline pallets/rolldown/rpc/src/lib.rs:85]: new rpc apis

File: pallets/rolldown/src/mock.rs [hunk 2]
```diff
@@ -151,6 +158,11 @@ impl rolldown::Config for Test {
 	type ChainId = messages::Chain;
 	type RightsMultiplier = ConstU128<1>;
 	type AssetAddressConverter = crate::MultiEvmChainAddressConverter;
+	type MerkleRootAutomaticBatchSize = ConstU128<10>;
+	type MerkleRootAutomaticBatchPeriod = ConstU128<25>;
+	type ManualBatchExtraFee = ConstU128<0>;
+	type TreasuryPalletId = TreasuryPalletId;
+	type NativeCurrencyId = NativeCurrencyId;
 }
```
Comment (mateuszaaa) [inline pallets/rolldown/src/mock.rs:166]: pallet_rolldown configuration parameters

File: pallets/sequencer-staking/src/lib.rs [hunk 3]
```diff
@@ -181,6 +181,14 @@ pub mod pallet {
...
 		StorageMap<_, Blake2_128Concat, (AccountIdOf<T>, T::ChainId), BalanceOf<T>, ValueQuery>;
 
+	#[pallet::storage]
+	pub type AliasAccount<T: Config> =
+		StorageMap<_, Blake2_128Concat, (AccountIdOf<T>, T::ChainId), AccountIdOf<T>, OptionQuery>;
+
+	#[pallet::storage]
+	pub type AliasAccountInUse<T: Config> =
+		StorageMap<_, Blake2_128Concat, AccountIdOf<T>, (), OptionQuery>;
+
```
Comment (mateuszaaa) [inline pallets/sequencer-staking/src/lib.rs:191]: Alias account for sequencer storages

File: pallets/rolldown/src/lib.rs [hunk 4]
```diff
@@ -241,15 +312,49 @@ pub mod pallet {
...
+		OptionQuery,
+	>;
+
+	#[pallet::storage]
+	#[pallet::unbounded]
+	/// For each supported chain stores:
+	/// - last batch id
+	/// - range of the reqeusts in last batch
+	pub type L2RequestsBatchLast<T: Config> =
+		StorageValue<_, BTreeMap<T::ChainId, (BlockNumberFor<T>, u128, (u128, u128))>, ValueQuery>;
```
Comment (mateuszaaa) [inline pallets/rolldown/src/lib.rs:365]: new storages that stores crated batches

**Review by iStrike7 — APPROVED** [review]

**Pull request merged**